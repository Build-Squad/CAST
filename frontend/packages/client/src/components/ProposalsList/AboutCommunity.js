import React from "react";
import Blockies from "react-blockies";

const CommunityMemberInfo = ({ name }) => {
  return (
    <div className="is-flex is-align-items-center py-3">
      <div className="is-flex pr-2">
        <Blockies seed={name} size={8} scale={4} className="blockies" />
      </div>
      <div className="subtitle has-text-grey small-text">{name}</div>
    </div>
  );
};

const RoleTitle = ({ role }) => {
  return (
    <div className="column is-12 pl-0 pb-1">
      <h6 className="subtitle small-text pb-2 pt-4">{role}</h6>
    </div>
  );
};

const AboutCommunity = ({
  textAbout = "",
  adminMembers = [],
  authorsMembers = [],
} = {}) => {
  return (
    <>
      <h1 className="title is-4 mb-5 is-hidden-mobile">About</h1>
      <p className="mb-5 has-text-grey">{textAbout}</p>
      <div className="divider py-2" />
      <RoleTitle role={"Admin"} />
      {adminMembers.map((item, index) => (
        <CommunityMemberInfo name={item.name} key={`member-${index}`} />
      ))}
      <div className="divider py-2" />
      <RoleTitle role={"Authors"} />
      {authorsMembers.map((item, index) => (
        <CommunityMemberInfo name={item.name} key={`member-${index}`} />
      ))}
    </>
  );
};
export default AboutCommunity;
