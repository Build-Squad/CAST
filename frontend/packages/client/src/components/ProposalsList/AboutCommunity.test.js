import React from "react";
import { cleanup, render } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import AboutCommunity from "./AboutCommunity";

let props = {};
afterEach(cleanup);

jest.mock("react-blockies", () => (props) => {
  const MockName = "blockies-component-mock";
  return <MockName {...props} />;
});

beforeAll(() => {
  props = {
    textAbout:
      "Gitcoin is the community of builders, creators, and protocols at the center of open web ecosystems. We build and fund Public Goods.",
    adminMembers: [
      {
        name: "nick-1.eth",
      },
      {
        name: "ceresstation-1.eth",
      },
    ],
    authorsMembers: [
      {
        name: "nick-2.eth",
      },
      {
        name: "ceresstation-2.eth",
      },
    ],
  };
});

afterAll(() => {
  jest.unmock("react-blockies");
});

describe("AboutCommunity UI Component", () => {
  it("renders correctly", () => {
    const component = renderer.create(
      <AboutCommunity
        textAbout={props.textAbout}
        items={props.items}
        adminMembers={props.adminMembers}
        authorsMembers={props.authorsMembers}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("AboutCommunity", () => {
  let rendered;
  beforeEach(() => {
    rendered = render(
      <AboutCommunity
        textAbout={props.textAbout}
        items={props.items}
        adminMembers={props.adminMembers}
        authorsMembers={props.authorsMembers}
      />
    );
  });
  it("should display 'About' title and about text", () => {
    const { getByText } = rendered;
    expect(getByText(props.textAbout)).toBeInTheDocument();
    expect(getByText("About")).toBeInTheDocument();
  });
  it("should display admin and authors", () => {
    const { getByText } = rendered;
    props.adminMembers.forEach(({ name }) => {
      expect(getByText(name)).toBeInTheDocument();
    });
    props.authorsMembers.forEach((member) => {
      expect(getByText(member.name)).toBeInTheDocument();
    });
  });
});

describe("AboutCommunity with no props", () => {
  let rendered;
  beforeEach(() => {
    rendered = render(<AboutCommunity />);
  });
  it("should display 'About' title only", () => {
    const { queryByText, getByText } = rendered;
    expect(queryByText(props.textAbout)).toBe(null);
    expect(getByText("About")).toBeInTheDocument();
  });
  it("should not display admin and authors", () => {
    const { queryByText } = rendered;
    props.adminMembers.forEach(({ name }) => {
      expect(queryByText(name)).toBe(null);
    });
    props.authorsMembers.forEach((member) => {
      expect(queryByText(member.name)).toBe(null);
    });
  });
});
