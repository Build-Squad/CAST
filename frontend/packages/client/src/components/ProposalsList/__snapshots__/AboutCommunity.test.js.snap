// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutCommunity UI Component renders correctly 1`] = `
Array [
  <h1
    className="title is-4 mb-5 is-hidden-mobile"
  >
    About
  </h1>,
  <p
    className="mb-5 has-text-grey"
  >
    Gitcoin is the community of builders, creators, and protocols at the center of open web ecosystems. We build and fund Public Goods.
  </p>,
  <div
    className="divider py-2"
  />,
  <div
    className="column is-12 pl-0 pb-1"
  >
    <h6
      className="subtitle small-text pb-2 pt-4"
    >
      Admin
    </h6>
  </div>,
  <div
    className="is-flex is-align-items-center py-3"
  >
    <div
      className="is-flex pr-2"
    >
      <blockies-component-mock
        className="blockies"
        scale={4}
        seed="nick-1.eth"
        size={8}
      />
    </div>
    <div
      className="subtitle has-text-grey small-text"
    >
      nick-1.eth
    </div>
  </div>,
  <div
    className="is-flex is-align-items-center py-3"
  >
    <div
      className="is-flex pr-2"
    >
      <blockies-component-mock
        className="blockies"
        scale={4}
        seed="ceresstation-1.eth"
        size={8}
      />
    </div>
    <div
      className="subtitle has-text-grey small-text"
    >
      ceresstation-1.eth
    </div>
  </div>,
  <div
    className="divider py-2"
  />,
  <div
    className="column is-12 pl-0 pb-1"
  >
    <h6
      className="subtitle small-text pb-2 pt-4"
    >
      Authors
    </h6>
  </div>,
  <div
    className="is-flex is-align-items-center py-3"
  >
    <div
      className="is-flex pr-2"
    >
      <blockies-component-mock
        className="blockies"
        scale={4}
        seed="nick-2.eth"
        size={8}
      />
    </div>
    <div
      className="subtitle has-text-grey small-text"
    >
      nick-2.eth
    </div>
  </div>,
  <div
    className="is-flex is-align-items-center py-3"
  >
    <div
      className="is-flex pr-2"
    >
      <blockies-component-mock
        className="blockies"
        scale={4}
        seed="ceresstation-2.eth"
        size={8}
      />
    </div>
    <div
      className="subtitle has-text-grey small-text"
    >
      ceresstation-2.eth
    </div>
  </div>,
]
`;
